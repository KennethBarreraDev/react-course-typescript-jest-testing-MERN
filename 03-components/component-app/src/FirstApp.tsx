//Using typescript types instead of proptypes
type NewMessage = {
    message: string,
    title: string
}

type Properties = {
    title?: string,
    subtitle: string
}

//Homework, return value from function
const returnMessage = (): NewMessage => {
    return {
        message: 'Hello world',
        title: 'A simple message'
    }
}

export const FirstApp = ({ title='Probe title', subtitle}: Properties) => {
    return (
        <>
            <h1 data-testid='test-title'>{title}</h1>
            <h2 data-testid='test-subtitle'>{subtitle}</h2>
            <h3>{JSON.stringify(returnMessage())}</h3>
        </>
    )
}

