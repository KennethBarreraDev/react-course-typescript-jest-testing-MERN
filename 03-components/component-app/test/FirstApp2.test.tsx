import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp.tsx"; // Importa tu componente correctamente

describe("FirstApp Component", () => {
  const subtitle = 'First paragraph';
  const title = 'This is a title';
  test("Should do match with snapshot", () => {
    const {container} = render(<FirstApp subtitle={subtitle}/>)
    expect(container).toMatchSnapshot()
  });

  test("Should render subtitle", () => {
    render(<FirstApp subtitle={subtitle}/>)
    expect(screen.getByText(subtitle)).toBeTruthy();
  });

  test("Should render title in h1 tag", () => {
    render(<FirstApp subtitle={subtitle} title={title}/>)

    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
  });
});
