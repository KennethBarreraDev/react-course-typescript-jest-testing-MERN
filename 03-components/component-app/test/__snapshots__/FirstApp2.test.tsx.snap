// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp Component Should do match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Probe title
  </h1>
  <h2
    data-testid="test-subtitle"
  >
    First paragraph
  </h2>
  <h3>
    {"message":"Hello world","title":"A simple message"}
  </h3>
</div>
`;
